pkg/util/cloudfoundry/garden_linux.go

```diff
@@ -0,0 +1,130 @@
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"code.cloudfoundry.org/garden"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/containers/providers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	// Needs ContainerImpl, so make sure it is init
+	_ "github.com/DataDog/datadog-agent/pkg/util/containers/providers/cgroup"
+)
+
+// GetGardenContainers returns the list of running containers from the local garden API
+func (gu *GardenUtil) GetGardenContainers() ([]garden.Container, error) {
+	return gu.cli.Containers(nil)
+}
+
+// ListContainers returns the list of running containers and populates their metrics and metadata
+func (gu *GardenUtil) ListContainers() ([]*containers.Container, error) {
+	gardenContainers, err := gu.GetGardenContainers()
+	if err != nil {
+		return nil, fmt.Errorf("error listing garden containers: %v", err)
+	}
+
+	var cList = make([]*containers.Container, len(gardenContainers))
+	handles := make([]string, len(gardenContainers))
+	for i, gardenContainer := range gardenContainers {
+		handles[i] = gardenContainer.Handle()
+	}
+	gardenContainerInfo, err := gu.cli.BulkInfo(handles)
+	if err != nil {
+		return nil, fmt.Errorf("error getting info for garden containers: %v", err)
+	}
+	gardenContainerMetrics, err := gu.cli.BulkMetrics(handles)
+	if err != nil {
+		return nil, fmt.Errorf("error getting metrics for garden containers: %v", err)
+	}
+
+	for i, handle := range handles {
+		infoEntry := gardenContainerInfo[handle]
+		if err := infoEntry.Err; err != nil {
+			log.Debugf("could not get info for container %s: %v", handle, err)
+			continue
+		}
+		metricsEntry := gardenContainerMetrics[handle]
+		if err := metricsEntry.Err; err != nil {
+			log.Debugf("could not get metrics for container %s: %v", handle, err)
+			continue
+		}
+		container := containers.Container{
+			Type:        "garden",
+			ID:          handle,
+			EntityID:    containers.BuildTaggerEntityName(handle),
+			State:       infoEntry.Info.State,
+			Excluded:    false,
+			Created:     time.Now().Add(-metricsEntry.Metrics.Age).Unix(),
+			AddressList: parseContainerPorts(infoEntry.Info),
+		}
+		cList[i] = &container
+	}
+
+	if err := providers.ContainerImpl().Prefetch(); err != nil {
+		return nil, fmt.Errorf("could not fetch container metrics: %s", err)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not get cgroups: %s", err)
+	}
+	for _, container := range cList {
+		if container.State != containers.ContainerActiveState || !providers.ContainerImpl().ContainerExists(container.ID) {
+			log.Debugf("Container %s not in state %s, skipping", container.ID[:12], containers.ContainerActiveState)
+			continue
+		}
+		limits, err := providers.ContainerImpl().GetContainerLimits(container.ID)
+		if err != nil {
+			log.Debugf("Cannot get limits for container %s: %s, skipping", container.ID[:12], err)
+			continue
+		}
+		container.SetLimits(limits)
+	}
+	err = gu.UpdateContainerMetrics(cList)
```

Comment:
This call is not necessary in itself as `Prefetch` was already done.
To avoid doing 2 `Prefetch` and 2 loops over containers, it might be best to mimic the changes that were done in `DockerUtil`.